ssl/record/ssl3_record.c

```diff
@@ -162,15 +162,14 @@ int ssl3_get_record(SSL *s)
              * Check whether this is a regular record or an SSLv2 style record.
              * The latter can only be used in the first record of an initial
              * ClientHello for old clients. Initial ClientHello means
-             * s->first_packet is set and s->server is true. The first record
-             * means we've not received any data so far (s->init_num == 0) and
-             * have had no empty records. We check s->read_hash and
-             * s->enc_read_ctx to ensure this does not apply during
-             * renegotiation.
+             * s->first_packet is set and s->server is true.  The first record
+             * means s->rlayer.is_first_record is true. Probably this is
+             * sufficient in itself instead of s->first_packet, but I am
+             * cautious. We check s->read_hash and s->enc_read_ctx to ensure
+             * this does not apply during renegotiation.
```

Comment:
(I believe `read_hash`/`enc_read_ctx` are indeed redundant, yeah. Likewise with the `s->first_packet` check. Though I'm not familiar with the 1.1.0 record-layer code so perhaps I'm missing something.)

This is a remarkably long-winded comment that, now that the logic is pretty straightforward, doesn't seem to add much. I'd probably just have written:

```
/*
 * The first record received by the server may be a V2ClientHello.
 */
if (s->server && RECORD_LAYER_is_first_record(&s->rlayer)
        && (p[0] & 0x80) && (p[2] == SSL2_MT_CLIENT_HELLO)) {
```

And maybe stuck the redundant checks in as `assert`s if they seemed interesting enough.